native/src/routeCalculationProgress.cpp

```diff
@@ -97,7 +98,34 @@ void RouteCalculationProgress::updateStatus(float distanceFromBegin, int directS
 	this->reverseSegmentQueueSize = reverseSegmentQueueSize;
 }
 
+float RouteCalculationProgress::getLinearProgressHH() {
+	float progress = 0;
+
+	for (int i = HHIteration::HH_NOT_STARTED; i <= HHIteration::DONE; i++) {
+		auto found = hhIterationPercent.find((HHIteration) i);
+		if (found != hhIterationPercent.end()) {
+			if (i == hhIterationStep) {
+				progress += hhCurrentStepProgress * (float) found->second; // current step
+				break;
+			} else {
+				progress += (float) found->second; // passed step
+			}
+		}
+	}
+
+	if (hhTargetsTotal > 0) {
+		progress = (100.0 * (float) hhTargetsDone + progress) / (float) hhTargetsTotal; // intermediate points
+	} else {
+		OsmAnd::LogPrintf(OsmAnd::LogSeverityLevel::Error, "Error: getLinearProgressHH() got hhTargetsTotal=0");
```

Comment:
Check cancel route and remove if redunant

Reply:
this logger should be kept until release version